modules/zivid/image.py

```diff
@@ -32,44 +41,27 @@ def save(self, file_path):
         This method will throw an exception if failing to save to the provided file_path.
 
         Args:
-            file_path: destination path
-
+            file_path: A pathlib.Path instance or a string specifying destination
         """
         self.__impl.save(str(file_path))
 
-    def to_array(self):
-        """Convert the image into a numpy array.
+    def copy_data(self):
+        """Copy image data to numpy array.
 
         Returns:
-            a numpy array
-
+            A numpy array containing color pixel data
         """
         self.__impl.assert_not_released()
         return numpy.array(self.__impl)
 
-    @property
-    def height(self):
-        """Return height (number of rows) of image.
-
-        Returns:
-            an integer
-
-        """
-        return self.__impl.height()
-
-    @property
-    def width(self):
-        """Return width (number of columns) of image.
-
-        Returns:
-            an integer
-
-        """
-        return self.__impl.width()
-
     def release(self):
         """Release the underlying resources."""
-        self.__impl.release()
+        try:
+            impl = self.__impl
+        except AttributeError:
+            pass
+        else:
+            impl.release()
```

Comment:
Good question. If you look up at the `__init__` method you will see that there is a chance for it to fail before `self.__impl` has been set. If then `__exit__` or `__del__` is called to clean up we would get an `AttributeError` raised from `release()` because `self.__impl` doesn't exist. 

Exceptions from the destructor are discarded, which is why it doesn't make unit tests red, but they will sometimes yield an ugly message in stderr. I saw this both during testing and in the wild when Amy was trying to do hand-eye, so I deployed this fix to our releasable classes.

Reply:
I see, good that you fixed it